### PR TITLE
Fix any broken links in the User Manual and rework lost links from 8.11

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context
- General cleanup
- Lost links from Gradle 8.11

### CODE REVIEW

File: platforms/documentation/docs/src/docs/userguide/redirects/dependency_downgrade_and_exclude.adoc
Comment (ov7a) [inline platforms/documentation/docs/src/docs/userguide/redirects/dependency_downgrade_and_exclude.adoc:1]: 💅 Would be nice to have the logic extracted, but duplicating  is fine for now

**Review by ov7a — APPROVED** [review]

File: platforms/documentation/docs/src/docs/userguide/jvm/groovy_plugin.adoc [hunk 1]
```diff
@@ -177,7 +177,7 @@ _Default value_: Not null
 +
 All Groovy source files of this source set. Contains only the `.groovy` files found in the Groovy source directories.
 
-These properties are provided by a convention object of type link:{groovyDslPath}/org.gradle.api.tasks.GroovySourceSet.html[GroovySourceSet].
+These properties are provided by a convention object of type link:{groovyDslPath}/org.gradle.api.tasks.SourceSet.html[GroovySourceSet].
```
Comment (cobexer) [inline platforms/documentation/docs/src/docs/userguide/jvm/groovy_plugin.adoc:180]: ❌ The link text no longer matches the target `GroovySourceSet` => `SourceSet`.

Also the deprecation of `GroovySourceSet` seems complicated, maybe the surrounding documentation needs updates as well?

> Using convention to contribute to source sets is deprecated. You can configure the groovy sources via the `GroovySourceDirectorySet` extension (e.g.
> `sourceSet.getExtensions().getByType(GroovySourceDirectorySet.class).setSrcDirs(...)`). This interface is scheduled for removal in Gradle 9.0.

File: platforms/documentation/docs/src/docs/userguide/jvm/scala_plugin.adoc [hunk 2]
```diff
@@ -216,7 +216,7 @@ _Default value:_ `[__projectDir__/src/__name__/scala]`.
 `allScala` — link:{javadocPath}/org/gradle/api/file/FileTree.html[FileTree] (read-only)::
 All Scala source files of this source set. Contains only the `.scala` files found in the Scala source directories. _Default value:_ non-null.
 
-These extensions are backed by an object of type link:{groovyDslPath}/org.gradle.api.plugins.scala.ScalaSourceSet.html[ScalaSourceSet].
+These extensions are backed by an object of type link:{javadocPath}/org/gradle/api/tasks/ScalaSourceSet.html[ScalaSourceSet].
```
Comment (cobexer) [inline platforms/documentation/docs/src/docs/userguide/jvm/scala_plugin.adoc:219]: 💅 `ScalaSourceSet` is actually deprecated, maybe the documentation here should be updated.

> Using conventions to contribute source sets is deprecated. You can configure the groovy sources via the `ScalaSourceDirectorySet` extension (e.g.
> `sourceSet.getExtensions().getByType(ScalaSourceDirectorySet.class).setSrcDirs(...)`). This interface is scheduled for removal in Gradle 9.0.

**Review by cobexer — CHANGES_REQUESTED** [review]

File: platforms/documentation/docs/src/docs/userguide/jvm/groovy_plugin.adoc (hunk 1 above)
Comment (lkasso) [reply platforms/documentation/docs/src/docs/userguide/jvm/groovy_plugin.adoc:180]: These pages are due for an update in a future PR.

File: platforms/documentation/docs/src/docs/userguide/jvm/scala_plugin.adoc (hunk 2 above)
Comment (lkasso) [reply platforms/documentation/docs/src/docs/userguide/jvm/scala_plugin.adoc:219]: These pages are due for an update in a future PR.

**Pull request merged**